iquip/apps/scheduler.py

```diff
@@ -30,6 +34,40 @@ def _dismiss_items(layout: Optional[QLayout] = None):
                 _dismiss_items(item.layout())
 
 
+def _run_thread_with_worker(worker: QObject):
```

Comment:
Nice idea to make it as a separate function.
However, please check that it is okay to
1. create `QThread` without `parent` (maybe that determines the thread where the new thread _lives_)
2. the `thread` object becomes not accessible after the function returns, and won't it make any issues (e.g., garbage collected)

In addition, I think this should also use camel case.

Reply:
For the camel case, this gives a pylint warning

```scheduler.py:37:0: C0103: Function name "_runThreadWithWorker" doesn't conform to snake_case naming style (invalid-name)```

Reply:
> For 1, I will add a `parent` argument to this function.

👍 

> For 2, the lines
> 
> ```
> worker.done.connect(thread.quit)
> worker.done.connect(worker.deleteLater)
> thread.finished.connect(thread.deleteLater)
> ```
> 
> does the work.

I was worrying about _unwanted_ garbage collection. 
As you experienced before, objects are garbage-collected when its reference is removed in Python.
For example, when we didn't assign an object to a variable, or the scope of a variable ended, the object is deleted by Python.
However in this case, as you said, `worker.done.connect(thread.quit)` would make a reference to `thread` from `worker` and `worker` is returned from the function, so it would be okay... I guess.

Reply:
> For the camel case, this gives a pylint warning
> 
> `scheduler.py:37:0: C0103: Function name "_runThreadWithWorker" doesn't conform to snake_case naming style (invalid-name)`

Oh, where did you run pylint? You should run it in the repository root directory, since the `.pylintrc` file is located there. Or you have to set the rc file manually when you run the command.

Reply:
> Oh, where did you run pylint? You should run it in the repository root directory, since the `.pylintrc` file is located there. Or you have to set the rc file manually when you run the command.

Can you tell me how I do that?

Reply:
Just go to the root directory `iquip` and run `pylint iquip`.
To designate an rcfile, run `pylint iquip --rcfile=.pylintrc`.
However, as the [documentation](https://pylint.pycqa.org/en/latest/user_guide/usage/run.html#command-line-options) says, it will automatically find `.pylintrc` file.

Reply:
Oh wait, I was confused. That was a function, not a method!
I am sorry for the confusion. I'm not sure which case (snake or camel) to use here...
Which do you prefer?

Reply:
I guess it is better to use the snake case, to have consistency with the function above!